packages/espressocash_app/lib/features/kyc_sharing/screens/phone_confirmation_screen.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:dio/dio.dart';
+import 'package:flutter/material.dart';
+
+import '../../../di.dart';
+import '../../../l10n/l10n.dart';
+import '../../../ui/bottom_button.dart';
+import '../../../ui/loader.dart';
+import '../../../ui/snackbar.dart';
+import '../services/kyc_service.dart';
+import '../utils/kyc_utils.dart';
+import '../widgets/kyc_page.dart';
+import '../widgets/kyc_text_field.dart';
+
+class PhoneConfirmationScreen extends StatefulWidget {
+  const PhoneConfirmationScreen({super.key});
+
+  static Future<bool> push(BuildContext context) => Navigator.of(context)
+      .push<bool>(
+        MaterialPageRoute(
+          builder: (context) => const PhoneConfirmationScreen(),
+        ),
+      )
+      .then((result) => result ?? false);
+
+  @override
+  State<PhoneConfirmationScreen> createState() =>
+      _PhoneConfirmationScreenState();
+}
+
+class _PhoneConfirmationScreenState extends State<PhoneConfirmationScreen> {
+  final _controller = TextEditingController();
+
+  bool get _isValid => _controller.text.length == 6;
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  Future<void> _handleConfirm() async {
+    final success = await runWithLoader<bool>(
+      context,
+      () async {
+        try {
+          final service = sl<KycSharingService>();
+          await service.verifyPhone(code: _controller.text);
+
+          return true;
+        } on Exception catch (exception) {
+          if (!mounted) return false;
+
+          final message = exception is DioException &&
```

Comment:
It should be updated in all the other places as well. Also, the error logic should be moved to the service itself, so that returns some business logic, e.g. `WrongCodeException`. In that case, presentation logic would be just about reacting to the error:

```dart
try {
  ...
} on WrongCodeException {
  // show 'Wrong verification code'
} on Exception {
  // try again later
}
```

Also, "Wrong verification code" message should be localized.